language/move-compiler/src/typing/core.rs

```diff
@@ -1703,3 +1728,23 @@ fn check_num_tvar_(subst: &Subst, loc: Loc, ty: &Type) -> bool {
         _ => false,
     }
 }
+//**************************************************************************************************
```

Comment:
I can't comment on non-touched lines (curse you github), so I'll leave it here.

You will need to handle contravariance with function types for subtyping. So you will need to do something like 
```
(Lambda(args1, ret1), Lambda(args2, ret2)) => {
  // ... arity checks
  match case {
    Subtyping => {
      // flip for variance in subtyping case
      subtype(args2, args1);
      subtype(ret1, ret2)
    }
    Join => {
      // no need to flip. a flip would be meaningless
      join(args1, args2)
      join(ret1, ret2)
    } 
  }
}
```

Reply:
Done. I added a concept of Variance for type names.